blueflood-core/src/main/java/com/rackspacecloud/blueflood/io/datastax/DAbstractMetricsRW.java

```diff
@@ -50,6 +48,18 @@ protected DAbstractMetricsRW(LocatorIO locatorIO) {
     protected abstract DAbstractMetricIO getIO( String rollupType, Granularity gran );
 
     /**
+     * Returns the Timer.Context object for timing the Datastax write
+     * @return
+     */
+    protected abstract Timer.Context startWriteTimer();
+
+    /**
+     * Returns the Timer.Context object for timing the waits of the write results
+     * @return
+     */
+    protected abstract Timer.Context startWaitResultsTimer();
+
+    /**
```

Comment:
You are right that there's a distinction here. I initially made ```startWaitResultsTimer``` to count the time it took for the whole loop to complete. But then I realized I'm just recording the same thing as the ```Batch Write Timer``` that's already existing. So I decided to record individual call to ```future.getUninterruptibly().all()```.